plugins/permission-backend-module-opa-wrapper/src/opa-client/opaClient.ts

```diff
@@ -88,12 +100,19 @@ export class OpaClient {
       });
       return opaPermissionsResponse.result;
     } catch (error: unknown) {
-      this.logger.error(
-        `An error occurred while sending the policy input to the OPA server: ${error}`,
-      );
-      throw new Error(
-        `An error occurred while sending the policy input to the OPA server: ${error}`,
-      );
+      const message = `An error occurred while sending the policy input to the OPA server:`;
+
+      if (error instanceof Error && error.name === 'FetchError') {
```

Comment:
What do you think of doing this instead here?

```ts
import fetch, { FetchError } from 'node-fetch';
//...
if (error instanceof FetchError)
```

Reply:
This line was the hardest to do. Essentially the mocking of `fetch` causes a lot of problems with actually testing a function with FetchError in it, as a mock node-fetch implementation will also make FetchError `undefined`, even with some `jest.requireActual` magic. That if statement is the most elegant solution I could come up with, surprisingly enough. To simplify it, fetch couldn't be mocked.

Reply:
I'd recommend adding a comment above explaining as such, but I prefer to ask you first.

Reply:
Sure thats fine with me 😄